React/Base/RCTUtils.m

```diff
@@ -347,6 +347,7 @@ BOOL RCTClassOverridesInstanceMethod(Class cls, SEL selector)
   if (error) {
     errorMessage = error.localizedDescription ?: @"Unknown error from a native module";
     errorInfo[@"domain"] = error.domain ?: RCTErrorDomain;
+    errorInfo[@"userInfo"] = error.userInfo;
```

Comment:
There's no guarantee that error.userInfo is JSON serializable, so this is a bit dangerous. You can make it safe by doing:

    errorInfo[@"userInfo"] = RCTJSONClean(error.userInfo);

But if all you need the userInfo for is to get the message, you probably don't need to do this at all (see my other comment).